### PR TITLE
fix: Optimization fix

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

This PR contains the fix for the optimizations. which includes changes in below area.
- App list 
- App Details
- Initial loading
- Module + ServerInfo handling
- Cluster list
- Other env Api call
- Chart store

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Tested by creating different scenarios including all the types of apps

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ClusterNodes/ClusterList.tsx [hunk 1]
```diff
@@ -195,28 +241,54 @@ export default function ClusterList({ imageList, isSuperAdmin, namespaceList }:
...
             </div>
         )
     }
 
-    if (loader) {
-        return <Progressing pageLoader />
+    const renderClusterList = (): JSX.Element => {
+        if (minLoader) {
+            return (
+                <div className="dc__overflow-scroll" style={{ height: 'calc(100vh - 116px)' }}>
```
Comment (shivani170) [inline src/components/ClusterNodes/ClusterList.tsx:267]: Inline style should be reduced

File: src/components/ClusterNodes/ClusterList.tsx [hunk 2]
```diff
@@ -195,28 +241,54 @@ export default function ClusterList({ imageList, isSuperAdmin, namespaceList }:
...
+                    <Progressing pageLoader />
+                </div>
+            )
+        } else if (noResults) {
+            return <ClusterNodeEmptyState actionHandler={clearSearch} />
+        } else {
+            return (
+                <div
+                    className="dc__overflow-scroll"
+                    style={{ height: `calc(${showTerminalModal ? '50vh - 125px)' : '100vh - 116px)'}` }}
```
Comment (shivani170) [inline src/components/ClusterNodes/ClusterList.tsx:277]: Remove Inline css

File: src/components/v2/index.tsx [hunk 3]
```diff
@@ -170,10 +168,9 @@ function RouterComponent({ envType }) {
 
     return (
         <React.Fragment>
-            {isLoading && <DetailsProgressing loadingText="Please wait…" size={24} fullHeight />}
-          {renderErrorScreen()}
-
-            {!isLoading && !errorResponseCode && (
+            {/* {isLoading && <DetailsProgressing loadingText="Please wait…" size={24} fullHeight />} */}
```
Comment (shivani170) [inline src/components/v2/index.tsx:171]: Is it not usable?

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 4]
```diff
@@ -189,7 +211,7 @@ export default function CDDetails() {
                     ) : (
                         <EmptyView
                             title="No deployments"
-                            subTitle={`No deployment history available for the ${environment?.environmentName} environment.`}
+                            subTitle={`No deployment history available for the ${selectedEnv?.environmentName} environment.`}
```
Comment (shivani170) [inline src/components/app/details/cdDetails/CDDetails.tsx:214]: Can extract this from constants

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 5]
```diff
@@ -249,70 +251,89 @@ export const Details: React.FC<DetailsType> = ({
...
-                        setExternalLinksAndTools(externalLinksAndTools)
-                        setAppDetailsLoading(false)
-                    })
-            } else {
-                setAppDetailsLoading(false)
-            }
-        } catch (error) {
-            if (error['code'] === 404 || appDetailsRef.current) {
+    const handleAppDetailsCallError = (error) => {
+        if (error['code'] === 404 || appDetailsRequestRef.current) {
```
Comment (shivani170) [inline src/components/app/details/appDetails/AppDetails.tsx:255]: Can extract error code from constants

File: src/components/Jobs/JobList/JobsList.tsx [hunk 6]
```diff
@@ -320,7 +320,7 @@ export default function JobsList() {
             )}
             {dataStateType === JobListViewType.LIST && (
                 <>
-                    <HeaderWithCreateButton headerName="Jobs" />
+                    <HeaderWithCreateButton headerName="Jobs" isSuperAdmin={true} />
```
Comment (shivani170) [inline src/components/Jobs/JobList/JobsList.tsx:323]: isSuperadmin will be always true?

File: src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx [hunk 7]
```diff
@@ -207,54 +215,60 @@ function EnvironmentSelectorComponent({isExternalApp, _init}: {isExternalApp: bo
...
+                    {!appDetails.deploymentAppDeleteRequest && (
+                        <button className="flex left small cta cancel pb-6 pt-6 pl-12 pr-12 en-2" onClick={showInfoUrl}>
+                            <LinkIcon className="icon-dim-16 mr-6 icon-color-n7" />
+                            Urls
+                        </button>
+                    )}
+                    {!showWorkloadsModal && (
+                        <button
+                            className="scale-workload__btn flex left cta cancel pb-6 pt-6 pl-12 pr-12 en-2 ml-6"
+                            onClick={() => setWorkloadsModal(true)}
```
Comment (shivani170) [inline src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx:237]: Avoid inline function

File: src/components/v2/appDetails/SyncError.component.tsx [hunk 8]
```diff
@@ -57,8 +57,9 @@ const SyncErrorComponent: React.FC<SyncErrorType> = ({ appStreamData, showApplic
                 </span>
                 {collapsed && (
                     <span className="cn-9 ml-24 w-80 dc__ellipsis-right">
-                        {isImagePullBackOff && !appDetails.externalCi && 'IMAGEPULLBACKOFF'}
-                        {conditions.length > 0 && ', '}
+                        {isImagePullBackOff &&
+                            !appDetails.externalCi &&
+                            `IMAGEPULLBACKOFF${conditions.length > 0 && ', '}`}
```
Comment (shivani170) [inline src/components/v2/appDetails/SyncError.component.tsx:62]: Is it working fine?

File: src/components/app/details/triggerView/ciMaterial.tsx [hunk 9]
```diff
@@ -253,41 +262,26 @@ export class CIMaterial extends Component<CIMaterialProps, CIMaterialState> {
...
+                    handleRegexInputValue={this.handleRegexInputValue}
+                    regexValue={this.state.regexValue}
+                    onCloseBranchRegexModal={this.props.onCloseBranchRegexModal}
+                    savingRegexValue={this.state.savingRegexValue}
+                />
+            )
+        } else if (this.props.showCIModal) {
+            return this.renderCIModal()
+        }
+        return <></>
```
Comment (shivani170) [inline src/components/app/details/triggerView/ciMaterial.tsx:285]: is it expected

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx [hunk 10]
```diff
@@ -48,15 +48,17 @@ export class TriggerCINode extends Component<TriggerCINodeProps> {
         this.props.history.push(this.getCIDetailsURL())
     }
 
+    hideDetails(status: string = '') {
+      return status === DEFAULT_STATUS.toLowerCase() || status === 'not triggered' || status === 'not deployed' || status ===''
```
Comment (shivani170) [inline src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:52]: not triggered &  not deployed can extract outside

File: src/components/app/list-new/AppList.tsx [hunk 11]
```diff
@@ -46,8 +47,8 @@ export default function AppList({ isSuperAdmin, appListCount, isArgoInstalled }:
     const [syncListData, setSyncListData] = useState<boolean>()
 
     // API master data
-    const [projectListRes, setProjectListRes] = useState({ result: [] })
-    const [environmentListRes, setEnvironmentListRes] = useState({ result: [] })
+    const [projectListRes, setProjectListRes] = useState([])
```
Comment (shivani170) [inline src/components/app/list-new/AppList.tsx:50]: Type is missing

File: src/components/app/list-new/AppList.tsx [hunk 12]
```diff
@@ -46,8 +47,8 @@ export default function AppList({ isSuperAdmin, appListCount, isArgoInstalled }:
...
+    const [environmentClusterListRes, setEnvironmentClusterListRes] = useState<EnvironmentClusterList>()
```
Comment (shivani170) [inline src/components/app/list-new/AppList.tsx:51]: Aren't we initializing with empty array?

File: src/components/charts/Charts.tsx [hunk 13]
```diff
@@ -13,7 +12,6 @@ export default function Charts() {
 
     return (
         <Switch>
-            <Route path={`${path}/deployments/:appId(\\d+)/env/:envId(\\d+)`} component={DeploymentDetail} />
```
Comment (shivani170) [inline src/components/charts/Charts.tsx:16]: removal is expected

File: src/components/common/header/PageHeader.tsx [hunk 14]
```diff
@@ -70,11 +70,11 @@ function PageHeader({
         setExpiryDate(+localStorage.getItem('clickedOkay'))
     }, [])
 
-    useEffect(() => {
-        if (!window._env_.K8S_CLIENT) {
-            getCurrentServerInfo()
-        }
-    }, [])
+    // useEffect(() => {
```
Comment (shivani170) [inline src/components/common/header/PageHeader.tsx:73]: Please remove

File: src/components/common/helpers/Helpers.tsx [hunk 15]
```diff
@@ -1216,4 +1218,14 @@ export const handleOnBlur = (e): void => {
 
 export const parsePassword = (password:string): string => {
     return password === DEFAULT_SECRET_PLACEHOLDER ? '' : password
+}
+
+export const reloadToastBody = () => {
+    return <UpdateToast
+        onClick={() => {
```
Comment (shivani170) [inline src/components/common/helpers/Helpers.tsx:1225]: Inline call back avoid

File: src/components/common/helpers/Helpers.tsx [hunk 16]
```diff
@@ -1216,4 +1218,14 @@ export const handleOnBlur = (e): void => {
...
+            window.location.reload()
+        }}
+        text="You are viewing an outdated version of Devtron UI."
```
Comment (shivani170) [inline src/components/common/helpers/Helpers.tsx:1228]: Extract Text from common

File: src/components/app/service.ts [hunk 17]
```diff
@@ -168,44 +179,54 @@ const gitTriggersModal = (triggers, materials) => {
...
-                                                  eventActionType: history.WebhookData.eventActionType,
-                                                  data: history.WebhookData.data,
-                                              }
-                                            : null,
-                                    }
-                                })
-                              : [],
+const processMaterialHistory = (material) => {
+    if (material.history) {
+        return material.history.map((history, indx) => {
```
Comment (milindk8) [inline src/components/app/service.ts:184]: index is mispelled

File: src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx [hunk 18]
```diff
@@ -65,119 +101,134 @@ function EnvironmentStatusComponent({ appStreamData }: { appStreamData: any }) {
...
-                                'status'
-                            ].toLowerCase()} dc__capitalize fw-6 fs-14 flex left`}
-                        >
-                            <span>{appDetails.additionalData['status']}</span>
-                            <figure
-                                className={`${appDetails.additionalData[
+    return (
+        <div>
+            {loadingDetails ? (
+                shimmerLoaderBlocks()
```
Comment (milindk8) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx:113]: break into small functions

File: src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx [hunk 19]
```diff
@@ -65,119 +101,134 @@ function EnvironmentStatusComponent({ appStreamData }: { appStreamData: any }) {
...
+                            </div>
+                            <div className=" fw-6 fs-14">
+                                {appDetails.appStoreChartName && <span>{appDetails.appStoreChartName}/</span>}
+                                {appDetails.appStoreAppName}({appDetails.appStoreAppVersion})
+                            </div>
+                            <div className="flex left">
+                                {notes && (
+                                    <div
+                                        className="details-hover flex cb-5 fw-6 cursor"
+                                        onClick={() => setShowNotes(true)}
```
Comment (milindk8) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx:196]: inline functions

File: src/components/ApplicationGroup/Details/TriggerView/BulkCDTrigger.tsx [hunk 20]
```diff
@@ -34,11 +34,17 @@ export default function BulkCDTrigger({
         appList.find((app) => !app.warningMessage) || appList[0],
     )
     const [unauthorizedAppList, setUnauthorizedAppList] = useState<Record<number, boolean>>({})
+    const abortControllerRef = useRef<AbortController>(new AbortController())
+
+    const closeBulkCDModal = (e) => {
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/ApplicationGroup/Details/TriggerView/BulkCDTrigger.tsx:39]: Done

File: src/components/app/details/appDetails/SourceInfo.tsx [hunk 21]
```diff
@@ -102,64 +109,105 @@ export function SourceInfo({
...
-                                {isHibernated ? 'Restore pod count' : 'Scale pods to 0'}
-                            </button>
+                {!loadingResourceTree && environment && (
+                    <>
+                        {!appDetails?.deploymentAppDeleteRequest && (
+                            <div style={{ marginLeft: 'auto' }} className="flex right fs-12 cn-9">
+                                {showUrlInfo && (
+                                    <button
+                                        className="cta cta-with-img small cancel fs-12 fw-6 mr-6"
+                                        onClick={(e) => showUrlInfo(true)}
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/app/details/appDetails/SourceInfo.tsx:119]: Done

File: src/components/app/details/triggerView/workflow.service.ts [hunk 22]
```diff
@@ -172,7 +225,8 @@ function addDimensions(workflows: WorkflowType[], workflowOffset: Offset, dimens
             (node) => node.type == WorkflowNodeType.CI || node.type == WorkflowNodeType.WEBHOOK,
         )
         ciNode.sourceNodes?.forEach((s, si) => {
-            let sourceNodeY = startY +
+            let sourceNodeY =
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/app/details/triggerView/workflow.service.ts:228]: Done

File: src/components/app/list/TriggerInfo.tsx [hunk 23]
```diff
@@ -112,16 +112,29 @@ export class TriggerInfoModal extends Component<TriggerInfoModalProps, TriggerIn
...
         else {
-            let selectedMaterial = this.state.materials.find(mat => mat.isSelected);
-            headerDescription = `Deployed on ${this.state.environmentName} at ${this.state.lastDeployedTime} by ${this.state.triggeredByEmail}`;
-            body = <div className="m-lr-0 flexbox">
-                <div className="material-list">
-                    <MaterialSource material={this.state.materials} selectMaterial={this.selectMaterial} />
-                </div>
-                <div className="select-material select-material--h450">
-                    <MaterialHistory material={selectedMaterial} pipelineName="" toggleChanges={this.toggleChanges} />
+            let selectedMaterial = this.state.materials.find((mat) => mat.isSelected)
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/app/list/TriggerInfo.tsx:115]: Done

File: src/components/app/service.ts (hunk 17 above)
Comment (shivani170) [reply src/components/app/service.ts:184]: Done

File: src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx (hunk 19 above)
Comment (shivani170) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx:196]: Done

File: src/components/ApplicationGroup/Details/TriggerView/BulkCITrigger.tsx [hunk 24]
```diff
@@ -56,12 +56,24 @@ export default function BulkCITrigger({
...
+        refreshMaterial: (
+            ciNodeId: number,
+            pipelineName: string,
+            materialId: number,
+            abortController?: AbortController,
+        ) => void
     } = useContext(TriggerViewContext)
+    const abortControllerRef = useRef<AbortController>(new AbortController())
+
+    const closeBulkCIModal = (evt) => {
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/ApplicationGroup/Details/TriggerView/BulkCITrigger.tsx:68]: As discussed this is expected.

File: src/components/v2/devtronStackManager/DevtronStackManager.tsx [hunk 25]
```diff
@@ -154,25 +152,8 @@ export default function DevtronStackManager({
 
     useEffect(() => {
         if (serverMode === SERVER_MODE.FULL) return
-        //Check for SERVER_MODE on every route change if Current SERVER_MODE is EA_ONLY
-        getVersionConfig()
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.tsx:158]: expected change as we don't need this anymore

File: src/components/ClusterNodes/ClusterList.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/ClusterList.tsx:267]: As discussed, we can keep this inline style.

File: src/components/ClusterNodes/ClusterList.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/ClusterNodes/ClusterList.tsx:277]: As discussed, we can keep this inline style.

File: src/components/ClusterNodes/ClusterList.tsx [hunk 26]
```diff
@@ -195,28 +241,54 @@ export default function ClusterList({ imageList, isSuperAdmin, namespaceList }:
...
+                >
+                    <div className="cluster-list-row fw-6 cn-7 fs-12 dc__border-bottom pt-8 pb-8 pr-20 pl-20 dc__uppercase">
+                        <div>Cluster</div>
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/ClusterNodes/ClusterList.tsx:280]: We'll be doing this in upcoming PR.

File: src/components/Jobs/JobList/JobsList.tsx (hunk 6 above)
Comment (vivek-devtron) [reply src/components/Jobs/JobList/JobsList.tsx:323]: Yes. this view is superAdmin specific

File: src/components/app/details/cdDetails/CDDetails.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/app/details/cdDetails/CDDetails.tsx:214]: We'll be covering this in upcoming PR

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx (hunk 10 above)
Comment (vivek-devtron) [reply src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:52]: done

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx [hunk 27]
```diff
@@ -48,15 +48,17 @@ export class TriggerCINode extends Component<TriggerCINodeProps> {
...
         let url = this.getCIDetailsURL()
         let status = this.props.status ? this.props.status.toLowerCase() : ''
-        let hideDetails =
-            status === DEFAULT_STATUS.toLowerCase() || status === 'not triggered' || status === 'not deployed'
-        if (hideDetails)
+        if (this.hideDetails(status))
             return (
                 <div className="dc__cd-trigger-status" style={{ color: TriggerStatus[status] }}>
-                    {this.props.status}
+                    {this.props.status ? this.props.status : 'not triggered'}
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:61]: done

File: src/components/app/list-new/AppList.tsx (hunk 12 above)
Comment (vivek-devtron) [reply src/components/app/list-new/AppList.tsx:50]: Not required

File: src/components/app/list/DevtronAppListContainer.tsx [hunk 28]
```diff
@@ -129,15 +129,15 @@ class DevtronAppListContainer extends Component<AppListProps, AppListState>{
...
+        
         getAppList(request, { signal: this.abortController.signal }).then((response) => {
             let view = AppListViewType.LIST;
             if (response.result.appCount === 0) {
                 if (isSearchOrFilterApplied) view = AppListViewType.NO_RESULT;
                 else view = AppListViewType.EMPTY;
             }
             let state = { ...this.state };
-            const apps = (response.result && !!response.result.appContainers) ? appListModal(response.result.appContainers) : []
+            const apps = (response.result && !!response.result.appContainers) ? appListModal(response.result.appContainers,this.props.environmentClusterList) : []
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/app/list/DevtronAppListContainer.tsx:148]: Done

File: src/components/app/list/appList.modal.ts [hunk 29]
```diff
@@ -20,18 +20,18 @@ export const buildInitState = (appListPayload): Promise<any> => {
...
 
-export const appListModal = (appList) => {
+export const appListModal = (appList, environmentClusterList) => {
     return appList.map((app) => {
         return {
             id: app.appId || 0,
             name: app.appName || 'NA',
-            environments: app.environments.map((env) => environmentModal(env)) || [],
-            defaultEnv: getDefaultEnvironment(app.environments),
+            environments: app.environments.map((env) => environmentModal(env,environmentClusterList)) || [],
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/app/list/appList.modal.ts:58]: Done

File: src/components/common/header/PageHeader.tsx (hunk 14 above)
Comment (vivek-devtron) [reply src/components/common/header/PageHeader.tsx]: Done

File: src/components/charts/Charts.tsx (hunk 13 above)
Comment (sohel-rp) [reply src/components/charts/Charts.tsx:16]: Yes

File: src/components/v2/appDetails/SyncError.component.tsx (hunk 8 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/SyncError.component.tsx:62]: Yes. Additional comma (,) was being displayed at the beginning when condition is not `IMAGEPULLBACKOFF`.

File: src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx (hunk 7 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx:237]: Existing. We'll be covering this in upcoming PR..

File: src/components/common/helpers/Helpers.tsx (hunk 15 above)
Comment (sohel-rp) [reply src/components/common/helpers/Helpers.tsx:1230]: Done

**Review by milindk8 — APPROVED** [review]

**Pull request merged**